src/main/java/org/assertj/core/api/AbstractBufferAssert.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import static org.assertj.core.error.ShouldBeEmpty.shouldBeEmpty;
+import static org.assertj.core.error.ShouldNotBeEmpty.shouldNotBeEmpty;
+import static org.assertj.core.error.buffer.ShouldBeFlipped.shouldBeFlipped;
+import static org.assertj.core.error.buffer.ShouldHaveLength.shouldHaveLength;
+import static org.assertj.core.error.buffer.ShouldHaveRemainingLength.shouldHaveRemainingLength;
+
+import java.nio.Buffer;
+
+/**
+ * Base class for all implementations of assertions for {@link Buffer}s.
+ *
+ * @param <SELF> the "self" type of this assertion class. Please read &quot;<a href="http://bit.ly/1IZIRcY"
+ *          target="_blank">Emulating 'self types' using Java Generics to simplify fluent API implementation</a>&quot;
+ *          for more details.
+ * @param <ACTUAL> the type of the "actual" buffer.
+ *
+ * @author Jean de Leeuw
+ */
+public abstract class AbstractBufferAssert<SELF extends AbstractBufferAssert<SELF, ACTUAL>, ACTUAL extends Buffer> extends AbstractAssert<SELF, ACTUAL> {
+
+  public AbstractBufferAssert(ACTUAL actual, Class<?> selfType) {
+    super(actual, selfType);
+  }
+
+  /**
+   * Verifies that the actual {@code Buffer} has been flipped.
+   *
+   * <b>Important:</b>
+   * Due to the limitations of the Buffer API there is, short of looping over the buffer and checking the content one
+   * by one, no way to check if the Buffer is actually flipped. It is impossible to make a distinction between
+   * a buffer that is filled to its capacity and a buffer that simply has not been flipped (or manual repositioning
+   * of the buffer). Due to these limitations, this method only checks if the position of the buffer equals 0.
+   * <p>
+   * Example:
+   *
+   * <pre><code class='java'>
+   * Buffer buffer = ByteBuffer.allocate(10);
+   *
+   * // this assertion succeeds ...
+   * buffer.flip();
+   * assertThat(buffer).isFlipped();
+   *
+   * // ... but this one fails as "allocate" does not flip the buffer.
```

Comment:
Let's not reuse buffer as it might be confusing as the previous `assertThat(buffer).isFlipped();` succeeded.
let's replace it with `assertThat(ByteBuffer.allocate(10)).isFlipped();`